crates/provider/src/provider/caller.rs

```diff
@@ -0,0 +1,42 @@
+use crate::ProviderCall;
+use alloy_json_rpc::{RpcParam, RpcReturn};
+use alloy_rpc_client::WeakClient;
+use alloy_transport::{RpcError, Transport, TransportErrorKind, TransportResult};
+use std::borrow::Cow;
+
+/// A caller that helper convert `RpcWithBlock` and `EthCall` into a `ProviderCall`.
+pub trait Caller<T, Params, Resp>: Send + Sync
```

Comment:
given we're now wonly using this for `EthCall`, I think we should make it more call-specific, eg make it accept `TransactionRequest`, block and overrides directly to allow implementations to directly call underlying evm with them.